Revisions Lecture 02B

sladenheim opened the pull request:
This PR adds:
- revisions to the Pandas section of the site chapter
- addition of Scikit-Learn content to the site chapter
- Rename the lecture to 02B-Pandas-Sklearn.qmd and modified references in _quarto.yml files

Closes #16

trgardos commented:
Hi @sladenheim -- a small request. did you rename the file in VS Code explorer UI? Apparently VS Code treats that as a `git rm` and then a new file called ..pandas-sklearn.qmd. I find that I have to do a `git mv oldfilename newfilename` in the terminal and then git continues to track it. Otherwise we lose the diff.

sladenheim commented:
That is strange because I definitely used the command: git mv 02B-Pandas.qmd 02B-Pandas-Sklearn.qmd. I just searched my terminal history to confirm this. I can always try to go back and interactive rebase this so that we don't lose the diff. Not sure why this didn't happen to begin with.

trgardos commented on ds701_book/02B-Pandas-Sklearn.qmd:
```diff
@@ -0,0 +1,610 @@
+1. Use the trained model to make predictions.
+1. Evaluate the performance of the model. 
+
+We will demonstrate the sklearn machine learning framework by working with the sklearn `LinearRegression` object. This object can be used to train a linear model that predicts continuous values. 
```
Makes sense to to use an example, but we do have the linear regrression chapter/lecture. Did you reconcile with that? I'm also reviewing now lecture 07 and it is introducing sklearn a bit too.

sladenheim replied:
I'm happy to make a separate lecture. I think that will make the essential tools chatper less bulky and easier to read.

I used linear regression because I had used it previously in my RCS tutorials. I referenced that we cover more theoretical/implementation details in the Linear Regression section. 

However, I think I should also look at the introduction that is made in lecture 07 and see if there is anything that could be moved from there to the sklearn chapter and then cross-reference from 07.

trgardos reviewed (commented):
Also, I wonder if we should just make sklearn it's own chapter. We can still reference more than one chapter in a lecture if we want to. Thoughts?

trgardos commented:
> That is strange because I definitely used the command: git mv 02B-Pandas.qmd 02B-Pandas-Sklearn.qmd. I just searched my terminal history to confirm this. I can always try to go back and interactive rebase this so that we don't lose the diff. Not sure why this didn't happen to begin with.

Weird. Usually it will show up as one file with an R next to it, rather than a delete and add. Did you do a force push btw? when i just did a `git pull` it shows the new branch and says "(forced update)". Not sure what that means.

sladenheim commented:
I think something must have gone wrong with the git mv. I checked out the earlier commit, switched to a new test branch and re-ran the git mv and the behavior is as expected. I will fix this and update the pull request.

Regarding the forced update: 

I pushed commits from my local issues/16-sklearn-branch to the remote(github) before your pull request for lecture 6. After  the lecture 6 content was merged into the main bracnh, I pulled the updated origin/main into my local main branch. I rebased my issues/16-sklearn-branch on top of my local main (which was up-to-date with the origin). This caused the remote issues/16-sklearn-branch to diverge from my local branch. I did a forced merge to github to get the branches up to date. I thought since I was the only person working on this branch it would not be an issue. I can avoid this in the future if it could cause problems.

trgardos commented:
> I think something must have gone wrong with the git mv. I checked out the earlier commit, switched to a new test branch and re-ran the git mv and the behavior is as expected. I will fix this and update the pull request.
> 
> Regarding the forced update:
> 
> I pushed commits from my local issues/16-sklearn-branch to the remote(github) before your pull request for lecture 6. After the lecture 6 content was merged into the main bracnh, I pulled the updated origin/main into my local main branch. I rebased my issues/16-sklearn-branch on top of my local main (which was up-to-date with the origin). This caused the remote issues/16-sklearn-branch to diverge from my local branch. I did a forced merge to github to get the branches up to date. I thought since I was the only person working on this branch it would not be an issue. I can avoid this in the future if it could cause problems.

In my experience force pushes can cause problems since i think it overwrites existing commits which cause problems in other branches. Usually there is a way to work around force pushes. happy to brainstorm if it comes up again.

trgardos commented:
is it ready to be merged?

sladenheim commented:
Yes we can merge this. I agree. We can brainstorm a better approach. I probably should have used the --force-with-lease option which only allows for a force push if nothing has been built on top of the remote branch (the documentation for this is here: https://git-scm.com/docs/git-push).
